specs/waku/v2/waku-relay.md

```diff
@@ -22,11 +30,34 @@ authors: Oskar Thorén <oskar@status.im>
 
 **Protocol identifier***: `/vac/waku/relay/2.0.0-beta2`
 
+## Security Requirements
+
+<!-- In this part, we analyze the security of the  `relay` protocol concerning data confidentiality, integrity, authenticity, and anonymity. This is to enable users of this protocol to make an informed decision about all the security properties that can or can not achieve by deploying a `relay` protocol.-->
+
+- **Message Publisher Anonymity**: This property indicates that no adversarial entity can link a published `Message` to its origin i.e., the peer. Note that this feature also implies the unlinkability of the publisher to its published topic ID, this is because  `Message` contains the topic IDs as well.
+
+- **Topic Subscriber Anonymity:** This feature stands for the inability of any adversarial entity from linking a peer to its subscribed topic IDs.
+
+- **Confidentiality**: The adversary should not be able to learn the data carried by the `relay` protocol.
+-  **Integrity**: This feature indicates that the data transferred by the `relay` protocol can not be tampered with by an adversarial entity without being detected.
+-  **Authenticity**: No adversary can forge data on behalf of a targeted peer and make it accepted by other peers as if the origin is the target.
+- **Spam resistant**: No adversary can flood the system with spam messages (i.e., publishing a large number of messages in a short amount of time).
+  
+<!-- TODO: more requirements can be added, but that needs further and deeper investigation-->
```

Comment:
For interest's sake, is this where the resilience of the underlying GossipSub protocol to attacks from malicious nodes come in? [This research article](https://www.researchgate.net/publication/342734066_GossipSub_Attack-Resilient_Message_Propagation_in_the_Filecoin_and_ETH20_Networks) lists a number of attacks and mitigation strategies, e.g. Sybil nodes affecting message propagation, mesh creation, etc. As far as I can tell many of the suggested mitigation strategies are implemented in nim-libp2p, though there may still be some vulnerabilities.

Reply:
> This research article lists a number of attacks and mitigation strategies, e.g. Sybil nodes affecting message propagation, mesh creation, etc. As far as I can tell many of the suggested mitigation strategies are implemented in nim-libp2p, though there may still be some vulnerabilities.

Thanks! will check it out

> is this where the resilience of the underlying GossipSub protocol to attacks from malicious nodes come in?

Yes, that is right

Reply:
> As far as I can tell many of the suggested mitigation strategies are implemented in nim-libp2p

Could not find the 6 scoring factors of the scoring function in nim-libp2p as described in the article, do you have any idea about their corresponding implementations in the codebase?